### PR TITLE
Build fixes and IDE instructions

### LINKED ISSUE #3338: [DOC] Instructions to setup IDEs
With some of the more recent changes my IDE setup stopped working and started to show compile errors.  This is because I had to select specific maven profiles to make it happy. The only reason I knew this was because I was following along with many of the build changes.  We really should have clear instructions about how to set up an IDE, including what maven profiles to enable/disable in specific situations.

### PR DESCRIPTION
**Pull request by gerashegalov**
- IDEA project setup instructions, surely more will be needed
- allow skipping scalastyle in repeated builds
- remove temporary `!buildver` profiles adding wrong  directories in IDEA
- rename spark311+-apache to spark311until320-apache to correctly reflect applicable shims

Fixes #3338

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

File: shims/pom.xml [hunk 1]
```diff
@@ -31,9 +31,50 @@
...
+        <!-- maven CLI and CI builds one at a time -->
+        <profile>
+            <id>with-classifier</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <modules><module>${shim.module.name}</module></modules>
+        </profile>
+
+        <!-- For easy switching in IDE -->
```
Comment (tgravescs) [inline shims/pom.xml:44]: I guess explicitly setting it makes ide happy?  can you see them all at once now?  perhaps a bit more explanation here

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -108,6 +108,32 @@ The version-specific directory names have one of the following forms / use cases
 - `src/main/302to312-cdh` contains code that applies to Cloudera CDH shims between 3.0.2 *inclusive*,
    3.1.2 *inclusive*
 
+
+### Setting up an Integrated Development Environment
+
+Our project currently uses `build-helper-maven-plugin` for shimming against conflicting definitions of superclasses
```
Comment (tgravescs) [inline CONTRIBUTING.md:114]: we could add a bit more here to clarify this pulls different source directories

**Comment by tgravescs** [comment]
overall looks good, couple nits

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -108,6 +108,32 @@ The version-specific directory names have one of the following forms / use cases
...
+modifications to IDEs' standard Maven import functionality.
+
+#### IntelliJ IDEA
+
+_Last tested with 2021.2.1 Community Edition_
+
+To start working with the project in IDEA is as easy as 
+[opening](https://blog.jetbrains.com/idea/2008/03/opening-maven-projects-is-easy-as-pie/) the top level (parent) 
+[pom.xml](pom.xml). 
+
```
Comment (andygrove) [inline CONTRIBUTING.md:125]: For me, it was also necessary to run one `mvn package -DskipTests` before I could run any tests. Without this, tests failed with `Could not find properties file rapids4spark-version-info.properties in the classpath`.

File: build/buildall [hunk 4]
```diff
@@ -18,12 +18,27 @@
...
+    -Dbuildver="$sparkver" \
+    -Drat.skip=true \
+    -DskipTests \
+    -Dmaven.javadoc.skip=true \
+    -Dskip \
+    -Dmaven.scalastyle.skip=true \
+    -pl aggregator -am
+done
+
+# install 301 without "skips" above
```
Comment (revans2) [inline build/buildall:43]: Nit: Not really true, we are still skipping the tests.

File: build/buildall [hunk 5]
```diff
@@ -18,12 +18,27 @@
...
+  304
+  311
+  311cdh
+  312
+  313
+  320
+"
+
+for sparkver in ${SPARK_SHIM_VERSIONS} ; do
+  mvn -U clean install \
```
Comment (revans2) [inline build/buildall:33]: nit: Might be good to have a comment about why we are skipping all of these things.

**Comment by tgravescs** [comment]
sid enote this passed databricks tests: https://blossom.nvidia.com/sw-gpu-spark-jenkins/job/rapids_databricks-for-developer/3/

File: shims/pom.xml (hunk 1 above)
Comment (gerashegalov) [reply shims/pom.xml:44]: I'll update, the main issue for IDEA is switching between differently named profiles that are activated by a single property value. I actually could not make it work even by adding as -D to the importer settings. So having consistently named profiles throughout submodules looks like a must here.

File: CONTRIBUTING.md (hunk 3 above)
Comment (gerashegalov) [reply CONTRIBUTING.md:125]: I believe this is a different IDE-unrelated regression #3461 that was introduced but I'll add wording about it for now if this workaround works consistently.

Comment (gerashegalov) [reply CONTRIBUTING.md:129]: I think we'll have to defer to #3461, I am not sure if this works generally. @andygrove feel free to add your instructions in a separate PR with the exact steps for your workaround.

**Comment by gerashegalov** [comment]
build

File: pom.xml [hunk 6]
```diff
@@ -448,7 +448,7 @@
                                 <configuration>
                                     <sources>
                                         <source>${project.basedir}/src/main/301until320-all/scala</source>
-                                        <source>${project.basedir}/src/main/301until320-nondb/scala</source>
+                                        <source>${project.basedir}/src/main/311until320-all/scala</source>
```
Comment (tgravescs) [inline pom.xml:451]: how is this working?  isn't is missing Spark30XShims?

Comment (gerashegalov) [reply pom.xml:451]: it does not work, it's an incorrect merge conflict resolution. and this build is not covered by pre-merge

**Comment by gerashegalov** [comment]
build

**Review by andygrove — APPROVED** [review]

File: pom.xml (hunk 6 above)
Comment (tgravescs) [reply pom.xml:452]: ah, ok.

**Review by tgravescs — APPROVED** [review]

**Comment by tgravescs** [comment]
@revans2 you good with updates?

**Review by revans2 — APPROVED** [review]
I have not had a chance to test that it has fixed all of my problems, but it is definitely a lot better than it was before.

**Comment by tgravescs** [comment]
ok going to merge then.

**Pull request merged**